cmake/FetchDependencies.cmake

```diff
@@ -83,9 +83,13 @@ set(DILIGENT_INSTALL_SAMPLES OFF CACHE BOOL "" FORCE)
 set(DILIGENT_INSTALL_FX OFF CACHE BOOL "" FORCE)
 set(DILIGENT_BUILD_SAMPLES OFF CACHE BOOL "" FORCE)
 set(DILIGENT_NO_RENDER_STATE_PACKAGER OFF CACHE BOOL "" FORCE)
-set(DILIGENT_NO_GLSLANG OFF CACHE BOOL "" FORCE)
 set(DILIGENT_NO_OPENGL ON CACHE BOOL "" FORCE)
 set(DILIGENT_NO_METAL ON CACHE BOOL "" FORCE)
+if(NC_RUNTIME_SHADER_COMPILATION)
+    set(DILIGENT_NO_GLSLANG OFF CACHE BOOL "" FORCE)
```

Comment:
We also want to have DILIGENT_NO_HLSL as part of this, or maybe just have _only_ DILIGENT_NO_HLSL in this conditional and always not build GLSLang at runtime - 
GLSLLang is the OpenGL/Vulkan/Spir-V toolchain, which is what we have in our codebase with all the legacy shaders.
HLSL is what Diligent recommends we use and what we will use going forward/what you have in `TestPipeline`

For runtime shader compiling Diligent can only use HLSL as the cross-API language. It can compile it to SPIRV or DX compatible bytecode. Diligent can runtime compile GLSL too, but can't convert GLSL to D3D compatible bytecode. Only HLSL goes both ways.

Any runtime shader compilation we do will be on shaders written in HLSL so we don't have to duplicate shaders, so we'll probably want to write our offline compiled shaders in HLSL as well

Reply:
We need GLSLang (at least for vulkan) even if the source language is HLSL. For runtime compilation, the GLSL compiler is still used, which calls a `GLSLangUtils::HLSLtoSPIRV()` function internally to convert it. It barfs if this is disabled.

Also, we need the HLSL option enabled to support the SPV_GOOGLE_hlsl_functionality1 extension with vulkan backend.